### ISSUE TITLE
knex.raw arguments in 0.8.6

### ISSUE THREAD

**Issue by wubzz** [tgriesser/knex#965]
I'm currently on hold with upgrading from 0.7.6 to 0.8.6 due to the knex events not being fixed in 0.8.6 yet, but I figured I could upgrade to test if theres anything else in my app that would need fixing.

Doing this, I found that the argument handling for `knex.raw` must have changed, and I'm just unsure whether this was intentional or not. (Nothing in changelog)

**0.8.6**
```javascript
knex.raw('SELECT schema_name FROM information_schema.schemata WHERE schema_name = ?', 'Test').toString();
//SELECT schema_name FROM information_schema.schemata WHERE schema_name = 'T'
//Also throws database error: 08P01 - bind message supplies 4 parameters, but prepared statement "" requires 1
```

**0.7.6**
```javascript
knex.raw('SELECT schema_name FROM information_schema.schemata WHERE schema_name = ?', 'Test').toString();
//SELECT schema_name FROM information_schema.schemata WHERE schema_name = 'Test'
```

Was this intentional?

**Comment by wubzz** [comment]
I should add that sending `['Test']` in 0.8.6 does work, but not if you send the string alone. Instead it appears to run `split` on the string..

**Comment by rhys-vdw** [comment]
Just a guess, I think it is possible to bind an array as a value. This
would cause ambiguity between applying an array as a binding and an array
of bindings.

If this is the case then it should error rather than splitting the string.

**Comment by wubzz** [comment]
@rhys-vdw I agree it should throw an error to easily detect broken parts of the apps.

**Comment by wubzz** [comment]
@rhys-vdw Would you agree that allowing either String or Array is the better option here? Much like nearly every other function in exposed in knex, such as the tablebuilder. (primary('id'), primary(['id'])

That way knex can be backwards-compatible in regards to this issue. It's a quick fix changing the argument to an Array if the input is a String. See: https://github.com/tgriesser/knex/blob/master/src/raw.js#L29

**Comment by elhigu** [comment]
Fixed in #1172
